grafana-plugin/src/components/Policy/EscalationPolicy.tsx

```diff
@@ -55,10 +52,26 @@ export interface EscalationPolicyProps extends ElementSortableProps {
   backgroundClassName?: string;
   backgroundHexNumber?: string;
   isSlackInstalled: boolean;
+  theme: GrafanaTheme2;
 }
 
 @observer
 class _EscalationPolicy extends React.Component<EscalationPolicyProps, any> {
+  private styles: ReturnType<typeof getEscalationPolicyStyles>;
+
+  constructor(props: EscalationPolicyProps) {
+    super(props);
+    this.styles = getEscalationPolicyStyles(props.theme);
+  }
+
+  componentDidUpdate(prevProps: Readonly<EscalationPolicyProps>): void {
+    if (prevProps.theme !== this.props.theme) {
+      // fetch new styles whenever the theme changes
+      this.styles = getEscalationPolicyStyles(this.props.theme);
+      this.forceUpdate();
+    }
+  }
```

Comment:
I think just as in functional components we could just define it in render like 
```
render() {
  const styles = getEscalationPolicyStyles(props.theme);
  ...
}
```

Reply:
agree, ideally styles should be just in render body and nowhere else

Reply:
Too many render functions, that means 1 line for each subrender where we're using the styles... not worth it, it takes 3 lines with cDU lifecycle.